arches/app/models/migrations/11725_make_tiledata_not_nullable.py

```diff
@@ -0,0 +1,18 @@
+# Generated by Django 5.2.dev20250109163842 on 2025-01-10 15:56
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
```

Comment:
This SQL looks like:
```sql
BEGIN;
--
-- Alter field data on tilemodel
--
ALTER TABLE "tiles" ALTER COLUMN "tiledata" SET DEFAULT '{}'::jsonb;
UPDATE "tiles" SET "tiledata" = '{}'::jsonb WHERE "tiledata" IS NULL; SET CONSTRAINTS ALL IMMEDIATE;
ALTER TABLE "tiles" ALTER COLUMN "tiledata" SET NOT NULL;
ALTER TABLE "tiles" ALTER COLUMN "tiledata" DROP DEFAULT;
COMMIT;
```

A db_default on this field didn't seem of much value, but I will circle back to #10958 to get moving on db defaults for primary key fields.

Reply:
This avoids having two representations of no data--`None` and `{}`.

Reply:
We could discuss whether to do the same thing for provisionaledits.

Reply:
Where this actually came up in practice is that while dev'ing #11596 I occasionally created corrupt tiles with data = None, and this caused failures all over where we depend on calling `.data.keys()`.

Reply:
Clarified over slack with Cyrus, repeating here: this doesn't represent an _intent_ to save any tiles with `{}` in the data column, this is just a something-better-than-nothing way to get `null=False` on this column, and then we can we add some other layer of protection against `{}` later, e.g. as a check constraint.